src/vivarium_gates_mncnh/components/propensity.py

```diff
@@ -0,0 +1,57 @@
+from itertools import combinations
+
+import numpy as np
+import pandas as pd
+from statsmodels.distributions.copula.api import GaussianCopula
+from vivarium import Component
+from vivarium.framework.engine import Builder
+
+from vivarium_gates_mncnh.constants import data_values
+
+
+class CorrelatedPropensities(Component):
+    """A component that takes components of any type as arguments and produces propensities
+    such that they are correlated according to the correlation values we define for each pair
+    of compoents in data_values. These propensities will be accessible as pipelines called
+    `<component_name>.correlated_propensity` in the simulation.
+    These propensities will not be used by any components as a result of creating these
+    pipelines, including the passed components, and it is the responsibility of each component
+    that wants to use these propensities to access them explicitly."""
+
+    ##############
+    # Properties #
+    ##############
+
+    def __init__(self, *component_names: str):
```

Comment:
Nit: I think a better way of doing this would be to have no argument to this component, and have a configuration key in the model spec:
```yaml
correlated_propensities:
    - "component_a"
    - "component_b"
    - ...
```